### ISSUE TITLE
Nuxt static site not reloading with local changes

### ISSUE THREAD

**Issue by jhull** [netlify/cli#655]
Nuxt Static site running through `netlify dev`.

Redirects and API calls work great (no more CORS!), but issue now is that any changes to files, and it doesn’t reload - the way it would with `npm run dev`. I have to stop the server, rebuild with `npm run build`, and then start up 'netlify dev` again.

I did look [here](https://community.netlify.com/t/netlify-dev-doesnt-reload/745) and [here](https://community.netlify.com/t/netlify-dev-live-is-not-hot-reloading-when-i-change-code/320), but those relate to `--live` and my issue is just straight up dev server.

I get the following error when I try to start nuxt in dev mode (`npm run dev`) before running `netlify dev`

> nuxt start

 WARN  vendor has been deprecated due to webpack4 optimization        18:11:02

 FATAL  No build files found in /Users/.../.nuxt/dist/server.
Use either `nuxt build` or `builder.build()` or start nuxt in development mode.

  Use either `nuxt build` or `builder.build()` or start nuxt in development mode.
  at VueRenderer._ready (node_modules/@nuxt/vue-renderer/dist/vue-renderer.js:4226:13)
  at async Server.ready (node_modules/@nuxt/server/dist/server.js:607:5)
  at async Nuxt._init (node_modules/@nuxt/core/dist/core.js:655:7)

 
   │                                                                   │
   │   ✖ Nuxt Fatal Error                                              │
   │                                                                   │
   │   Error: No build files found in                                  │
   │   /Users/.../.nuxt/dist/server.               │
   │   Use either `nuxt build` or `builder.build()` or start nuxt in   │
   │   development mode.

**Comment by landjoanthan** [comment]
I'm having similar issues.
As a workaroud I've found that if I first run `nuxt --port 3333` (or anything other than the default 3000) and then run `ntl dev` it'll kinda work – my dev server runs on 3333 and functions are available on the default `localhost:34567`. Not sure about redirects or your webpack issues, though. Hope this helps somehow.

**Comment by narrative1st** [comment]
How do you run that first one? trying `npm run dev --port 3333` and `nuxt --port 3333` and can't seem to get this working.

**Comment by landjoanthan** [comment]
```js
// package.js
"scripts": {
    "dev": "nuxt --port 3333",
...
```
Works for me. What error do you get?

**Comment by narrative1st** [comment]
It was working about an hour ago, managed to figure out that's what you meant prior to your post...but now I can't get `localhost:8888` to start up. I get these endless dots. https://github.com/netlify/cli/issues/657#issuecomment-568165817

**Comment by landjoanthan** [comment]
OK, just realized the correct approach is to set Dev to run `nuxt`:
```tmol
# netlify.toml
[dev]
  command = "nuxt"
```

Not sure why `nuxt build` is the default, but now everything seems to work as you'd expect, with HMR on `localhost:3000` and functions at `locahost:3000/.netlify/functions/`.

**Comment by RaeesBhatti** [comment]
We have robust support for Nuxt in recent versions. Running  `netlify dev` in your Nuxt project should automatically configure and run everything necessary or prompt you.
If the automatic configuration doesn't work, you can specify `command` in `netlify.toml` `[dev]` section to run Nuxt development server if the auto chosen command is problematic. And specify `targetPort` to connect to Nuxt development server if it is using port other than `3000`.

Please let us know if this is still an issue for you!

**Comment by jhull** [comment]
@RaeesBhatti 

Upgraded to the latest, and everything stopped working 🙁

(latest version of nuxt and netlify-cli)

I have this in `netlify.toml`:
```
[dev]
  command = "nuxt"
  port = 8888
  targetPort = 3000
```

And used to be able to run everything off of `http://frontend.localhost:8888` but now that doesn't work, and only `http://frontend.localhost:3000` works (which means all my CORS setup no longer works locally anymore) - and then I have the endless dots that continue (is that a feature now?)

**Comment by RaeesBhatti** [comment]
@jhull Can you please elaborate! What is the error you're receiving? Or how is it not working?

**Comment by jhull** [comment]
Nuxt frontend, Laravel API backend, Netlify Functions to bridge the two. Held back updating CLI until I saw this latest...but now I’m getting CORS issues when all I changed was update Netlify CLI (worked great for past 5 months)

Things I need to know:

a) are the endless dots a signal that ntl dev is working right? (Older version didn’t do this)

b) with the above `toml` settings, I access my local site through `http://frontend.localhost:8888` right? Because it fails now, yet I can access through `http://frontend.localhost:3000` — but then my CORS settings fail

c) are there any demos/working models of this same setup in action, I can’t find anything anywhere
